### PR TITLE
Fix pwa configuration, current release only works with default values

### LINKED ISSUE #383: sw.js and workbox-*.js do not end up in public folder
# Summary

When defining following config in my `next.config.js` the files `sw.js` and `workbox-*.js` do not end up in the public folder as explained in the README.md that states if using Nextjs 9 or newer the files will be there. Instead I do see those files end up in the `.next` folder.

My config looks like:

```js
module.exports =   withPWA({
  pwa: {
    dest: "public"
  }
});
```

# Versions

- `next-pwa`: 5.5.5
- `next`: 12.2.4

# How To Reproduce

```shell
$ yarn build
yarn run v1.22.19
$ next build
info  - Loaded env from /Users/marco/code/priv/nextjs-blog/.env
info  - SWC minify release candidate enabled. https://nextjs.link/swcmin
Generated 76 documents in .contentlayer
info  - Linting and checking validity of types  
> [PWA] Compile server
> [PWA] Compile client (static)
> [PWA] Auto register service worker with: /Users/marco/code/priv/nextjs-blog/node_modules/next-pwa/register.js
> [PWA] Service worker: /Users/marco/code/priv/nextjs-blog/.next/sw.js
> [PWA]   url: /sw.js
> [PWA]   scope: /
> [PWA] Compile server
[BABEL] Note: The code generator has deoptimised the styling of /private/var/folders/lr/ctz75x8s1r39qp_y7flj6p2m0000gp/T/6f95f768c8708c926b774c8de4734e97/sw.js as it exceeds the max of 500KB.
info  - Creating an optimized production build  
info  - Compiled successfully
info  - Collecting page data  
info  - Generating static pages (192/192)
info  - Finalizing page optimization 
```

# Expected Behaviors

The files end up in `public`.

# Screenshots

N.A.

# Additional Context

Also notice the BABEL Note in the log above. Maybe it makes sense to switch from Babel to `vercel/ncc` to bundle the service worker.

### PR DESCRIPTION
**Pull request by marcofranssen**
resolves #383

@shadowwalker could you make a `5.5.6` release including this fix?

I have tested the changes on my nextjs project using.

```shell
cd next-pwa
yarn link
cd ../my-blog
yarn link next-pwa
yarn build
```

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -20,11 +20,12 @@ module.exports = (nextConfig = {}) => {
...
-        config: { distDir = '.next', pwa = {}, pageExtensions = ['tsx', 'ts', 'jsx', 'js', 'mdx'], experimental = {} }
+        isServer,
+        dir,
+        config: { distDir = '.next', pageExtensions = ['tsx', 'ts', 'jsx', 'js', 'mdx'], experimental = {} },
       } = options
-
-      let basePath = options.config.basePath
-      if (!basePath) basePath = '/'
+      const basePath = options.config.basePath || '/'
+      const { pwa } = nextConfig
```
Comment (marcofranssen) [inline index.js:28]: This is the real fix, the rest is a bit of code cleanup.

File: index.js [hunk 2]
```diff
@@ -20,11 +20,12 @@ module.exports = (nextConfig = {}) => {
         webpack,
         buildId,
         dev,
-        config: { distDir = '.next', pwa = {}, pageExtensions = ['tsx', 'ts', 'jsx', 'js', 'mdx'], experimental = {} }
```
Comment (marcofranssen) [inline index.js:23]: the pwa part in here resulted in not taking the pwa config at all.

File: index.js [hunk 3]
```diff
@@ -20,11 +20,12 @@ module.exports = (nextConfig = {}) => {
...
+        isServer,
+        dir,
+        config: { distDir = '.next', pageExtensions = ['tsx', 'ts', 'jsx', 'js', 'mdx'], experimental = {} },
       } = options
-
-      let basePath = options.config.basePath
-      if (!basePath) basePath = '/'
+      const basePath = options.config.basePath || '/'
+      const { pwa = {} } = nextConfig
```
Comment (marcofranssen) [inline index.js:28]: This is the real fix, the rest is a bit of code cleanup.

**Comment by Lukasdotcom** [comment]
This is almost the same thing as #381. Except that it also does some code cleanup.

**Comment by ymchun** [comment]
https://github.com/shadowwalker/next-pwa/blob/9d28c3be6aa6cc082a2ca87ba61117fff57ded8f/index.js#L337

the comment here explained why `delete pwaNextConfig.pwa` is added. since Next.js will issue a warning for unknown properties of nextConfig

**Comment by marcofranssen** [comment]
@shadowwalker could we take this fix to be able to release a patch for the current version? In parallel we should start on introducing the breaking change following best practices as discussed in #381.

File: index.js (hunk 3 above)
Comment (levifry) [reply index.js:28]: Thank you!!! It works 🎉

**Comment by levifry** [comment]
Need this PR approved ASAP. Package is broken until approved

**Pull request closed without merging**